src/provider.toit

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2025 Toit contributors.
+// Use of this source code is governed by an MIT-style license that can be found
+// in the LICENSE file.
+
+import gpio
+import i2c
+import sensors.providers
+
+import .bme68x as bme68x
+
+NAME ::= "toit.io/bme68x"
+MAJOR ::= 1
+MINOR ::= 0
+
+class Sensor_
+    implements
+      providers.TemperatureSensor-v1
+      providers.HumiditySensor-v1
+      providers.PressureSensor-v1:
+  sda_/gpio.Pin? := null
+  scl_/gpio.Pin? := null
+  i2c_/i2c.Bus? := null
+  device_/i2c.Device? := null
+  sensor_/bme68x.Bme68x? := null
+
+  constructor --sda/int --scl/int --address/int:
+    is-exception := true
+    try:
+      sda_ = gpio.Pin sda
+      scl_ = gpio.Pin scl
+      i2c_ = i2c.Bus --sda=sda_ --scl=scl_
+      device_ = i2c_.device address
+      sensor_ = bme68x.Bme68x device_
+      sensor_.on
+      is-exception = false
+    finally:
+      if is-exception:
+        if device_: device_.close
```

Comment:
What happens if `sensor_.on` throws? Maybe you should close that one too in that case. Maybe there is nothing to do.